torchao/dtypes/nf4tensor.py

```diff
@@ -11,10 +11,6 @@
 from torch import Tensor
 from torch.distributed.device_mesh import DeviceMesh
 from torch._prims_common import make_contiguous_strides_for
-from torchao.dtypes.utils import (
```

Comment:
https://github.com/pytorch/ao/pull/360 consolidate `_implements` and `_ATEN_OP_OR_TORCH_FN_TABLE` but it breaks torchtune. revert for now to unblock torchtune quickly

Reply:
Do you know how exactly this breaks torchtune? is it a versioning thing between saved models and this new model?

Reply:
the error is `TypeError: nf4_detach() missing 1 required positional argument: 'args'`. So there is something incompatiable around with NF4 around `_ATEN_OP_OR_TORCH_FN_TABLE[func]`

the errors shows up when people start training in TorchTune for the 1st time

Reply:
@jerryzh168  any thoughts on why this is happening, otherwise are you okay to undo your changes?

Reply:
@drisspg to add a bit more to what @weifengpy already said, the full stack trace is [here](https://gist.github.com/ebsmothers/154bafd723d6d58f3d61a3b42e3e913f).

Reply:
<img width="791" alt="Screenshot 2024-06-21 at 11 22 11 AM" src="https://github.com/pytorch/ao/assets/32754868/ccb3f410-2162-4205-b922-7ec484118172">


I think there is something weird going on with the torch_function dispatch, where the args are becoming the arg name for aten.. If someone wants to track down how this should work and why this is I am for it but otherwhise I will approve to unblock